File: leetcode/53. Maximum Subarray/MaximumSubarray.go
func maxSubArray(nums []int) int {
	sum, maxSum := 0, nums[0]

	for _, num := range nums {
		sum += num
		maxSum = max(maxSum, sum)
		if sum < 0 {
			sum = 0
		}
	}
	return maxSum
}

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

File: leetcode/53. Maximum Subarray/MaximumSubarray.cpp
class Solution
{
public:
    int maxSubArray(vector<int> &nums)
    {
        int sum = 0, maxSum = nums[0];
        for (const int num : nums)
        {
            sum += num;
            maxSum = max(sum, maxSum);
            if (sum < 0)
            {
                sum = 0;
            }
        }
        return maxSum;
    }
};